internal/server/users.go

```diff
@@ -463,6 +469,18 @@ func (s *Server) deleteUserHandler() http.HandlerFunc {
 		if roles.IsSuperAdmin {
 			err = s.Store.DeleteUserByID(r.Context(), id)
 		} else {
+			// Only super-admins can delete super-admins
+			targetUser, err := s.Store.GetUserByID(r.Context(), id)
```

Comment:
Hmm, this should probably be done in a transaction so the user can't change, right? For example, the user could be normal, so we don't 403, but they're quickly updated to be a super admin, and we don't know about that so we delete the super admin anyways.